app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,21 @@
+#
+class SessionsController < ApplicationController
+  def new
+  end
+
+  def create
```

Comment:
Assignment Branch Condition size for create is too high. Need to refactor this action to make it simpler.  This could be done by creating a couple private methods...

    def create
      if authenticate
        session[:user_id] = user.id
        redirect_to user, notice: 'Logged in!'
      else
        flash.now.alert = 'Invalid email or password'
        render 'new'
      end
    end

    #... 

    private

    def authenticate
      return user.authenticate(params[:user][:password]) if user
      false
    end

    def user
      @_user ||= User.find_by_email(params[:user][:email])
    end